multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java

```diff
@@ -45,13 +40,30 @@ public ProcessType getProcessType(DelegateExecution execution, boolean required)
         }
     }
 
-    protected String getServiceId(DelegateExecution execution) {
-        String serviceId = VariableHandling.get(execution, Variables.SERVICE_ID);
-        if (serviceId == null) {
+    public ProcessType getProcessTypeFromExecution(DelegateExecution execution) {
+        String processDefinitionId = execution.getProcessDefinitionId();
+        int serviceIdEndIndex = processDefinitionId.indexOf(':');
+
+        if (serviceIdEndIndex == -1) {
             String correlationId = VariableHandling.get(execution, Variables.CORRELATION_ID);
             throw new SLException(MessageFormat.format(Messages.UNKNWON_SERVICE_ID_FOR_PROCESS_0, correlationId));
         }
-        return serviceId;
+
+        String serviceId = processDefinitionId.substring(0, serviceIdEndIndex);
```

Comment:
This code is confusing at first sight. Maybe extracting it in a method with a name like `getServiceIdFromProcessDefinition(String processDefinitionId)` will help readability.